cmd/gaia/init/testnet.go

```diff
@@ -182,67 +200,115 @@ func testnetWithConfig(config *cfg.Config, cdc *codec.Codec) error {
 			return err
 		}
 
-		// Gather gentxs folder
+		// gather gentxs folder
 		err = writeFile(fmt.Sprintf("%v.json", nodeDirName), gentxsDir, txBytes)
 		if err != nil {
 			_ = os.RemoveAll(outDir)
 			return err
 		}
 	}
 
-	// Generate empty genesis.json
+	if err := initGenFiles(cdc, chainID, accs, genFiles); err != nil {
+		return err
+	}
+
+	if err := collectGenFiles(cdc, config, chainID, monikers, nodeIDs, valPubKeys); err != nil {
+		return err
+	}
+
+	fmt.Printf("Successfully initialized %d node directories\n", numValidators)
+	return nil
+}
+
+func initGenFiles(
+	cdc *codec.Codec, chainID string, accs []app.GenesisAccount, genFiles []string,
+) error {
+
+	numValidators := viper.GetInt(flagNumValidators)
 	appGenState := app.NewDefaultGenesisState()
 	appGenState.Accounts = accs
+
 	appGenStateJSON, err := codec.MarshalJSONIndent(cdc, appGenState)
 	if err != nil {
 		return err
 	}
+
 	genDoc := types.GenesisDoc{
 		ChainID:    chainID,
 		AppState:   appGenStateJSON,
 		Validators: nil,
 	}
 
-	// Save all genesis.json files
+	// generate empty genesis files for each validator and save
 	for i := 0; i < numValidators; i++ {
 		if err := genDoc.SaveAs(genFiles[i]); err != nil {
 			return err
 		}
 	}
 
+	return nil
+}
+
+func collectGenFiles(
+	cdc *codec.Codec, config *cfg.Config, chainID string,
+	monikers, nodeIDs []string, valPubKeys []crypto.PubKey,
+) error {
+
+	outDir := viper.GetString(flagOutputDir)
+	numValidators := viper.GetInt(flagNumValidators)
```

Comment:
same, let's not use viper this way, these should be inputs to the function and viper should be used only in the cobra commands

Reply:
k just realized you're only re-organizing here, and it was already this way - so probably okay to just leave for this PR - however it would still be great if you wanted to address this